pymc3/plots/posteriorplot.py

```diff
@@ -47,10 +49,23 @@ def plot_posterior_predictive_glm(trace, eval=None, lm=None, samples=30, **kwarg
     if "c" not in kwargs and "color" not in kwargs:
         kwargs["c"] = "k"
 
+    plotting_fn = _plot_multitrace if isinstance(trace, MultiTrace) else _plot_inferencedata
+    plotting_fn(trace, eval, lm, samples, kwargs)
+    plt.title("Posterior predictive")
+
+
+def _plot_multitrace(trace, eval, lm, samples, kwargs):
     for rand_loc in np.random.randint(0, len(trace), samples):
         rand_sample = trace[rand_loc]
         plt.plot(eval, lm(eval, rand_sample), **kwargs)
         # Make sure to not plot label multiple times
         kwargs.pop("label", None)
 
-    plt.title("Posterior predictive")
+
+def _plot_inferencedata(trace, eval, lm, samples, kwargs):
+    trace_df = trace.posterior.to_dataframe()
+    for rand_loc in np.random.randint(0, len(trace_df), samples):
+        rand_sample = trace_df.iloc[rand_loc]
+        plt.plot(eval, lm(eval, rand_sample), **kwargs)
+        # Make sure to not plot label multiple times
+        kwargs.pop("label", None)
```

Comment:
These two functions have a lot of duplicated lines; I think they can be merged into one by checking `if isinstance(trace, MultiTrace)` at the beginning of the function (or just before) and casting the `InferenceData` `to_array` (I think this is the name of the function but you can check on ArviZ website) instead of to a dataframe. 
After that, the handling should be the same as you're dealing with numpy arrays in both cases

Reply:
if we cast to `array` then I think we wouldn't be able to access the different parameters (e.g. `'Intercept'` or `'x'`), which appear in `lm`. Each element here is a dict in the multitrace case:
```python
> /home/mgorelli/pymc3-dev/pymc3/plots/posteriorplot.py(61)_plot_multitrace()
-> plt.plot(eval, lm(eval, rand_sample), **kwargs)
(Pdb) type(rand_sample)
<class 'dict'>
(Pdb) rand_sample
{'x': 1.0, 'Intercept': 1.0}
```

Reply:
Ah right, I forgot the whole trace was given here, and not only `trace["y"]` for instance. But then, wouldn't `trace.posterior.to_dataframe().to_dict()` get the format we want? That way we'd need only one plotting function

Reply:
sure - I think this'd be _slightly_ more expensive, but arguably it's worth it for the sake of much simpler code